src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb

```diff
@@ -550,28 +561,63 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
+                    tList.Add(MakeLineContinuationTrivia(GetText(1)))
+                    If Here > 1 Then
+                        tList.Add(MakeWhiteSpaceTrivia(GetText(Here - 1)))
+                    End If
+                    Dim comment As SyntaxTrivia = ScanComment()
+                    If Not CheckFeatureAvailability(Feature.CommentsAfterLineContinuation) Then
+                        comment = DirectCast(comment.SetDiagnostics({ErrorFactory.ErrorInfo(ERRID.ERR_CommentsAfterLineContinuationNotAvailable1,
+                                                                                                    New VisualBasicRequiredLanguageVersion(Feature.CommentsAfterLineContinuation.GetLanguageVersion()))}), SyntaxTrivia)
+                    End If
+                    tList.Add(comment)
+                    ch = Peek()
+                    atNewLine = IsNewLine(ch)
                 Else
-                    Exit While
+                    ' We have a Line Continuation without comment but have 0 or more spaces after _, so process as V15.5
+                    If Not CanGet(Here) Then
+                        Return False
+                    End If
+                    atNewLine = IsNewLine(Peek(Here))
+                    If Not atNewLine Then
+                        Return False
+                    End If
+                    tList.Add(MakeLineContinuationTrivia(GetText(1)))
+                    ' Leading Whitespace on line after _
+                    If Here > 1 Then
+                        tList.Add(MakeWhiteSpaceTrivia(GetText(Here - 1)))
+                    End If
                 End If
-            End While
-
-            ' Line continuation is valid at the end of the
-            ' line or at the end of file only.
-            Dim atNewLine = IsNewLine(ch)
-            If Not atNewLine AndAlso CanGet(Here) Then
-                Return False
-            End If
-
-            tList.Add(MakeLineContinuationTrivia(GetText(1)))
-            If Here > 1 Then
-                tList.Add(MakeWhiteSpaceTrivia(GetText(Here - 1)))
+            Else
+                ' We don't have a space or ' but we might have an EOF after _ and that is not an error
+                ' This case if different then above because we don't want to skip whitespace after _
+                If CanGet(Here) Then
+                    ch = Peek(Here)
+                End If
+                atNewLine = IsNewLine(ch)
+                If Not atNewLine AndAlso CanGet(Here) Then
+                    ' If we get here we have an error, return trivia is Nothing
+                    Return False
+                End If
+                ' If we get here the line ends in valid newline or EOF and there is no whitespace after _
+                tList.Add(MakeLineContinuationTrivia(GetText(1)))
             End If
 
-            If atNewLine Then
+            If atNewLine AndAlso CanGet() Then
```

Comment:
> AndAlso CanGet() [](start = 24, length = 17)

I am not sure why adding new condition ```AndAlso CanGet()``` is the right thing to do. Could you please elaborate?

Reply:
I don't think this was addressed yet.

---
In reply to: [213416380](https://github.com/dotnet/roslyn/pull/28327#discussion_r213416380) [](ancestors = 213416380)

Reply:
You can't call SkipLineBreak(ch, 0) if CanGet() is false and in at least one test atNewLine = True and CanGet is False. This can happen if there is a vbLf instead of a vbCrLf at the end of a line at the end of a file.
```
            If atNewLine AndAlso CanGet() Then
                Dim newLine = SkipLineBreak(ch, 0)
```

Reply:
There is something strange about this, when I run the test single threaded ch is always vbNullChar but when I run all the tests at once ParseWarningDirective_LineContinuation2V16 fails with ch =vbLf. I don't see anything I changed that could cause nondeterministic behavior.